explore/explore_internal.h

```diff
@@ -26,75 +26,78 @@ namespace exploration
   };
 
   // uniform random between 0 and 1
-  inline float uniform_random_merand48(uint64_t initial)
+  inline float uniform_random_merand48_advance(uint64_t& initial)
   {
     initial = CONSTANT_A * initial + CONSTANT_C;
     int_float temp;
     temp.i = ((initial >> 25) & 0x7FFFFF) | BIAS;
     return temp.f - 1;
   }
 
- template<typename It>
-  int generate_epsilon_greedy(float epsilon, uint32_t top_action, It pdf_first, It pdf_last, std::random_access_iterator_tag /* pdf_tag */)
+  // uniform random between 0 and 1
+  inline float uniform_random_merand48(uint64_t initial) { return uniform_random_merand48_advance(initial); }
```

Comment:
what is going on between uniform_random_merand48_advance and uniform_random_merand48?

Reply:
One advances the random state.  Other does not.

Reply:
is that the state of the seed or something like that? interesting

Reply:
One updates the random state and other one does not.